core/modules/layout/layout.context.admin.inc

```diff
@@ -201,7 +340,14 @@ function layout_context_add_form_submit($form, &$form_state) {
   if (!empty($handler->is_new)) {
     $handler->is_new = FALSE;
     $handler->storage = TRUE;
-    $layout->setContexts($handler->name, $handler);
+    if ($form_state['clicked_button']['#name'] == 'context_add_button') {
```

Comment:
In other places, we're checking `#array_parents` rather than `#name`, could we use all one approach or the other? I have a preference for `#name`, since it's less nesting.